content/english/debugging/e3-burrows-wheeler-transform.md

```diff
@@ -15,19 +15,19 @@ The algorithm is relatively straightforward and easy to understand.
 2. Sort the rotations in lexicographical order, ascending.
 3. Take the last column of the sorted transform; this is the output.
 
-![Burrows-Wheeler Transform](../resources/e3-01.png)
+![Burrows-Wheeler Transform](../resources/e3-01.png "The 3 steps of a Burrows Wheeler Transform")
 
 In the diagram above, we use the string `banana` as an example. The `\0` is a symbol used to denote the end of the string (as in the null terminator character in C), but is lexicographically the LARGEST character when compared to the others.
 
 An efficient way to implement the BWT is using a data structure called a suffix array. To create a suffix array, we first take all suffixes of an input string, and give it a numbered label. We then sort the suffixes lexicographically.
 
-![Suffix Arrays](../resources/e3-02.png)
+![Suffix Arrays](../resources/e3-02.png "Steps of creating a suffix array")
 
 Using `Nuevo\0` as an example, we first take the suffixes of it and list them out. We then sort them in lexicographic order, with `\0` being the null terminating character as before and acting as the largest character. Finally, we take the index of the first letter of each suffix, and put them into an array. For instance, the index of the letter `N` in `Nuevo\0` is `0`, while the index of `e` is `2`. Since they are alphabetically the smallest, they go into slots 0 and 1 of the array, respectively. 
 
 We can think of suffixes as rotations for the Burrows-Wheeler transform! All we need to do is append the *prefix* that we "chopped off" to the end of the string. So, `evo\0` becomes `evo\0Nu` when we want to think of the suffix as a rotation. Then, if we do this for all the suffixes, we get the sorted rotations we want in the Burrows-Wheeler transform.
 
-![Suffixes to Rotations](../resources/e3-03.png)
+![Suffixes to Rotations](../resources/e3-03.png "Table of sorting rotations from suffixes for the Burrows-Wheeler transform")
 
 A key note is that we don't actually need to store the rotations because the index in the suffix array gives us that information already. If `suffix_array[i] == 0`, then we know that the rotation we're looking at is `Nuevo\0`. If `suffix_array[i] == 3`, then the rotation we're looking at is `vo\0Nue`. You can think of it as us rotating `N` spaces to the right in the case `suffix_array[i] == N`. The illustration above demonstrates this.
 

```

Comment:
On line 46, can the tab character in the code snippet be replaced with 4 spaces?